src/main/java/io/quarkus/search/app/fetching/QuarkusIO.java

```diff
@@ -74,16 +80,34 @@ public void close() throws Exception {
 
     @SuppressWarnings("resource")
     public Stream<Guide> guides() throws IOException {
+        return Stream.concat(
+                localGuides(),
+                quarkiverseGuides());
+    }
+
+    private Stream<Guide> localGuides() throws IOException {
         return guideDirectories()
                 .flatMap(uncheckedIO(guidesDirectory -> Files.list(guidesDirectory.path)
                         .filter(path -> {
                             String filename = path.getFileName().toString();
-                            return !filename.startsWith("_") && !FilenameUtils.getBaseName(filename).equals("README")
+                            return !filename.startsWith("_") && !FilenameUtils.getBaseName(filename).equals(
+                                    "README")
                                     && FilenameUtils.isExtension(filename, "adoc");
                         })
                         .map(path -> parseGuide(guidesDirectory, path))));
     }
 
+    private Stream<Guide> quarkiverseGuides() throws IOException {
+        return Stream.concat(
+                quarkiverseDirectories()
+                        .flatMap(quarkiverse -> QuarkiverseMetadata.parseYamlMetadata(quarkiverse.path, quarkiverse.version)
+                                .createQuarkiverseGuides()),
+                quarkiverseLegacyDirectories()
+                        .flatMap(quarkiverse -> QuarkiverseMetadata
+                                .parseYamlLegacyMetadata(quarkiverse.path, quarkiverse.version)
+                                .createQuarkiverseGuides()));
+    }
+
     @SuppressWarnings("resource")
```

Comment:
At this point, wouldn't it make more sense to use the YAML metadata to drive the whole process? I.e. create one Guide per entry found in the YAML metadata?

Reply:
This would require filtering the metadata YAML when we include it in QuarkusIOSample, though.

Reply:
yup 🙈